src/client/datascience/jupyter/liveshare/hostJupyterServer.ts

```diff
@@ -254,4 +272,223 @@ export class HostJupyterServer extends JupyterServerBase implements INotebookSer
         );
         return { info: launchInfo, changedKernel };
     }
+
+    public async connect(launchInfo: INotebookServerLaunchInfo, cancelToken?: CancellationToken): Promise<void> {
```

Comment:
Moved all of this code from a base class into this class.
Base class was not required (I believe it was used in liveshare scenario)